File: src/app/models/team-info.ts
export interface TeamInfo {
  id: string;
  name: string;
  fullName: string;
  path: string;  // Logo path
  founded: number;
  stadium: {
    name: string;
    capacity: number;
    location: string;
    image?: string;
  };
  colors: {
    primary: string;
    secondary: string;
  };
  history: string;
  achievements: {
    leagueTitles: number;
    cupTitles: number;
    otherTitles?: { [key: string]: number };
  };
  website?: string;
  socialMedia?: {
    facebook?: string;
    twitter?: string;
    instagram?: string;
  };
} 

File: backend/src/main/java/com/gfl/api/model/TeamInfo.java
package com.gfl.api.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Entity
@Table(name = "team_info")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class TeamInfo {

    @Id
    private String id;

    private String name;

    private String fullName;

    private Integer founded;

    @Column(columnDefinition = "TEXT")
    private String history;

    private String website;

    @Embedded
    private Stadium stadium;

    @Embedded
    private Colors colors;

    @Embedded
    private Achievements achievements;

    @Embedded
    private SocialMedia socialMedia;
}
